cognite/client/data_classes/workflows.py

```diff
@@ -1230,6 +1232,36 @@ def _load_trigger(cls, data: dict) -> Self:
         raise NotImplementedError()
 
 
+class WorkflowTriggerDataModelingQuery(CogniteObject):
```

Comment:
Can you subclass `Query` instead?

Reply:
Query has `parameters` and `cursors` that we do not want to expose.

Reply:
If its okay with you, I'll push a commit with a suggestion here (that we'll make a `_QueryCore` with `with_` and `select`, that both query classes can inherit from.

Reply:
At second thought, why not something like this:

```py
class WorkflowTriggerDataModelingQuery(Query):
    def __init__(
        self,
        with_: dict[str, ResultSetExpression],
        select: dict[str, Select],
    ) -> None:
        # parameters and cursors are not supported:
        super().__init__(with_, select_, parameters=None, cursors=None)
```